core/blockchain.go

```diff
@@ -433,34 +433,29 @@ func ExecuteBlockEphemerallyBor(
 	}
 
 	blockLogs := ibs.Logs()
-	var stateSyncReceipt *types.ReceiptForStorage
+	stateSyncReceipt := &types.Receipt{}
```

Comment:
Use
```var stateSyncReceipt *types.Receipt```
Otherwise it will always pass this check
```		if stateSyncReceipt != nil {```

Reply:
Okay, This will create other problems, Need to think something else.

Reply:
This is for reference : https://github.com/maticnetwork/turbo-bor/blob/master/core/blockchain.go#L184
I guess now it makes sense why we used this flow in old version.